### ISSUE TITLE
Filepath built by babel-plugin-preval produces "Module not found" when required.

### ISSUE THREAD

**Issue by agstover** [zeit/next.js#8487]
# Bug report

## Describe the bug

I'm basing my work very closely off of this [NextJS + MDX blog boilerplate](https://github.com/lorenseanstewart/nextjs-mdx-blog-kit/blob/master/utils/page-list.js). Essentially I am using the [babel-plugin-preval](https://github.com/kentcdodds/babel-plugin-preval/blob/master/README.md) package to deep walk a directory in /pages, getting the filepaths before compile, and then trying to

`const {meta} = require('../' + filepath)`

so I can get the meta data. The preval runs successfully. I get an array of all the file paths which I can log to console before I attempt to require them. When I try to do the require, however, I get an error:

`
Error: Cannot find module './pages/learn/hvac/begin/how-to-fix-a-sink/step-1.mdx'
    at webpackContextResolve (/home/agstover/Desktop/code/kinetica/www/.next/server/static/development/pages/_app.js:3124:11)
    at webpackContext (/home/agstover/Desktop/code/kinetica/www/.next/server/static/development/pages/_app.js:3118:11)
    at buildLearnNode (/home/agstover/Desktop/code/kinetica/www/.next/server/static/development/pages/_app.js:3185:72)
    at /home/agstover/Desktop/code/kinetica/www/.next/server/static/development/pages/_app.js:3268:85
    at Array.map (<anonymous>)
    at _callee$ (/home/agstover/Desktop/code/kinetica/www/.next/server/static/development/pages/_app.js:3267:31)
    at tryCatch (/home/agstover/Desktop/code/kinetica/www/.next/server/static/development/pages/_app.js:1389:40)
    at Generator.invoke [as _invoke] (/home/agstover/Desktop/code/kinetica/www/.next/server/static/development/pages/_app.js:1615:22)
    at Generator.prototype.<computed> [as next] (/home/agstover/Desktop/code/kinetica/www/.next/server/static/development/pages/_app.js:1441:21)
    at asyncGeneratorStep (/home/agstover/Desktop/code/kinetica/www/.next/server/static/development/pages/_app.js:716:24)
    at _next (/home/agstover/Desktop/code/kinetica/www/.next/server/static/development/pages/_app.js:738:9) {
`

I have tried this with the most recent version of Next as well as 8.0.1. I have found some StackOverflow issues which suggest Webpack can't handle dynamic imports and they have to be prefixed w/a static path which I am doing. Also, if I hard code in the exact same file path, the require works.

I can't understand how what I'm doing is so different from the blog example I'm following that this would not work. 

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

These are the relevant functions, all in separate files:

```
export function getCourses() {
    const filepaths = preval`
        const fs = require('fs')
        const path = require('path')
        console.log("PREVAL-ING")
        const getAllFiles = dir =>
        fs.readdirSync(dir).reduce((files, file) => {
            const name = path.join(dir, file);
            const isDirectory = fs.statSync(name).isDirectory();
            return isDirectory ? [...files, ...getAllFiles(name)] : [...files, name];
        }, [])
        module.exports = getAllFiles('./pages/learn')
    ` || []
    return Promise.resolve(filepaths)
}

```
```
const buildLearnNode = filepath => {
    const node = buildIds(filepath)
    console.log("THIS IS FILEPATH", '../' + filepath)
    const result = require('../' + filepath)
    console.log("RESULT", result)
    return Object.assign({}, node, {
        Component,
        title: meta.title || null
    })
}
```
```
const learnEngine = async () => {
    let filepaths
    try {
        filepaths = await getCourses()
        console.log("DIRECTLY AFTER PREVAL", filepaths)
    } catch(err) {
        console.log("There was an error getting learn files", err)
    }
    const nodes = filepaths.map(filepath => buildLearnNode(filepath))
    const tree = arrayToTree(nodes)
    console.log("TREE", tree)
    return tree
}
```
And then I'm using the learnEngine() in _app page in getInitialProps()

```
  static async getInitialProps({ Component, ctx }) {
    let pageProps = {};

    const learnData = await learnEngine()
    console.log('AFTER', learnData)
    if (Component.getInitialProps) {
      pageProps = await Component.getInitialProps(ctx);
    }

    const propsObj = Object.assign(
      {},
      { ...pageProps }
    );

    return { ...propsObj };
  }
```
## Expected behavior

Expect files to be successfully required.

## System information

- OS: Ubuntu 18.04
- Browser (if applies) Brave Browser
- Version of Next.js: 8.0.1 and 9.0.4

**Comment by Timer** [comment]
Per the issue template, general help/questions belong in [our Spectrum community](https://spectrum.chat/next-js). Thanks!

**Comment by agstover** [comment]
I've posted this in Spectrum three times and it hasn't gotten any attention. What do I need to do to get some guidance on this?

**Comment by agstover** [comment]
For anyone that finds this in the future, I found a solution. My best guess as to what the issue is is that when Webpack sees a require('../some/path' + something) it bundles up '../some/path' ahead of time. Methinks maybe if the directory is too large or if the file is too deep, it won't do this. My filepath was pretty long '../pages/learn/hvac/begin/how-to-fix-a-sink/step-1.mdx'. I stripped that path down to just 'hvac/begin/how-to-fix-a-sink/step-1.mdx' and made the require(`../pages/learn/${filepath}) and it now works.
